byway/src/main/webapp/routepage.js

```diff
@@ -49,27 +63,139 @@ function calcRoute(directionsService, directionsRenderer, start, end) {
   });
 }
 
+/** Get the new list of stops from datastore onload */
+function getStops(){
+  const stopList = document.getElementById('stop-list');
+
+  if(stopList != null){
+    stopList.innerText = ""; // clear list
+  }
+
+  fetch('/api/stop')
+  .then(response => response.json())
+  .then((stopsResponse) => {
+    stopsResponse.forEach((stop)=>{
+      stops.push(stop);
+      var btn = document.createElement('button');
+      btn.id = stop.id;
+      btn.innerText = stop.placename;
+      btn.setAttribute("class", "btn rec-btn");
+      btn.addEventListener("click", function() {
+        deleteFromStops(stop); 
+      });
+     stopList.appendChild(btn);
+    })
+  })
+}
+
+/** Get the new list of stops locally */
+function getStopsList(){
```

Comment:
Not sure what the difference between `getStopsList()` and `getStops()` is. I see one gets from the server and other doesn't, but rather than clear them all and then add `n+1` stops, consider adding just the one stop by seeing if its already included in local array. If not, add the one new button, or else remove the one said button.

Reply:
Yes, this was brought up as a suggestion from Landon. My only issue is that what happens to the placement if I remove the button, does everything get shifted? This way was just clearer for me to implement.

Reply:
If it is easier to replace all items, then let's just do that for the time being. It's not uncommon to redraw an entire UI element once the underlying data change, though it can be inefficient. We can always optimize this later.

Reply:
Okay, sounds good.